From 43284298ed919470f12b5e1fff5f843700075397 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:37:18 -0400
Subject: [ethernet] i40e: Fix an issue with displaying IPv4 FD filters

Message-id: <1409658042-22689-78-git-send-email-sassmann@redhat.com>
Patchwork-id: 90271
O-Subject: [RHEL7.1 PATCH 077/281] i40e: Fix an issue with displaying IPv4 FD filters
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Anjali Singhai Jain <anjali.singhai@intel.com>

The flow spec coming in for IPv4 filters is IP_USER_FLOW, which
needed some more info to be communicated up above in order for it
to be displayed correctly.

Change-ID: Ia968238e0d7c4c4df12908ba81f0c4501280f3ec
Signed-off-by: Anjali Singhai Jain <anjali.singhai@intel.com>
Signed-off-by: Catherine Sullivan <catherine.sullivan@intel.com>
Tested-by: Kavindya Deegala <kavindya.s.deegala@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 7d54eb2c66f9e1c2704fb9ff982fe31a6d1623e8)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_ethtool.c b/drivers/net/ethernet/intel/i40e/i40e_ethtool.c
index db57cce..4d206d1 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_ethtool.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_ethtool.c
@@ -1189,6 +1189,12 @@ static int i40e_get_ethtool_fdir_entry(struct i40e_pf *pf,
   return -EINVAL;
 
  fsp->flow_type = rule->flow_type;
+ if (fsp->flow_type == IP_USER_FLOW) {
+  fsp->h_u.usr_ip4_spec.ip_ver = ETH_RX_NFC_IP4;
+  fsp->h_u.usr_ip4_spec.proto = 0;
+  fsp->m_u.usr_ip4_spec.proto = 0;
+ }
+
  fsp->h_u.tcp_ip4_spec.psrc = rule->src_port;
  fsp->h_u.tcp_ip4_spec.pdst = rule->dst_port;
  fsp->h_u.tcp_ip4_spec.ip4src = rule->src_ip[0];
-- 
1.7.1